feat(console): only use one save button in SMTP Settings

doncicuto opened the pull request:
In issue #5613 @hifabienne has proposed to split the notification settings using three sidemenu points:

- Notifications
- SMTP Provider
- SMS/Phone Provider

In this new content distribution, admins no longer have two save buttons (one for policy, one for smtp settings).

The following video shows the proposed behavior:

https://github.com/zitadel/zitadel/assets/30386061/fb0fa432-7829-49d5-89b0-6fc3904e9e5f

Closes #5613

### Definition of Ready

- [X] I am happy with the code
- [X] Short description of the feature/issue is added in the pr description
- [X] PR is linked to the corresponding user story
- [X] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [X] No debug or dead code
- [X] My code has no repetitions
- [X] Critical parts are tested automatically
- [X] Where possible E2E tests are implemented
- [X] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [X] Functionality of the acceptance criteria is checked manually on the dev system.

doncicuto commented:
@hifabienne I don't want to waste your limited time, so if @peintnermax hasn't started to review this PR maybe you can pause the review on this one and I may cherry pick code from this PR and add it to my, somehow epic, PR for SMTP templates, testing... that way SMTP settings will be changed only once and therefore admins won't be confused with frequent changes to the console. 

Also I owe myself a review to @peintnermax for https://github.com/zitadel/zitadel/pull/6612 :smile so I don't want to bother him twice (thanks @peintnermax for your patience and kind reviews)

peintnermax commented:
> @hifabienne I don't want to waste the team's limited time, so if @peintnermax hasn't started to review this PR maybe you can pause the review on this one and I may cherry pick code from this PR and add it to my, somehow epic, PR for SMTP templates, testing... that way SMTP settings will be changed only once and therefore admins won't be confused with frequent changes to the console.
> 
> Also I owe myself a review to @peintnermax for #6612 :smile so I don't want to bother him twice (thanks @peintnermax for your patience and kind reviews)

Hi @doncicuto Thx a lot for your contributions! Don't worry to much about it.. two PRs are fine 👌
Will review this PR asap.. 👍

peintnermax commented on console/src/app/modules/policies/notification-sms-provider/notification-sms-provider.component.ts:
```diff
@@ -0,0 +1,166 @@
+  public fileNotificationProvider!: DebugNotificationProvider.AsObject;
+
+  public smsProvidersLoading: boolean = false;
+  public logProviderLoading: boolean = false;
+  public fileProviderLoading: boolean = false;
```
those are not used atm.. lets create a single loading variable and render the spinner on `true` 👍

peintnermax commented on console/src/app/modules/policies/notification-sms-provider/notification-sms-provider.component.ts:
```diff
@@ -0,0 +1,166 @@
+export class NotificationSMSProviderComponent {
+  @Input() public serviceType!: PolicyComponentServiceType;
+  public smsProviders: SMSProvider.AsObject[] = [];
+  public logNotificationProvider!: DebugNotificationProvider.AsObject;
+  public fileNotificationProvider!: DebugNotificationProvider.AsObject;
```
those are not needed anymore as they are moved to the other component

peintnermax commented on console/src/app/modules/policies/notification-sms-provider/notification-sms-provider.component.ts:
```diff
@@ -0,0 +1,166 @@
+        this.toast.showError(error);
+      });
+
+    this.logProviderLoading = true;
+    this.service
+      .getLogNotificationProvider()
+      .then((logNotificationProvider) => {
+        this.logProviderLoading = false;
+        if (logNotificationProvider.provider) {
+          this.logNotificationProvider = logNotificationProvider.provider;
+        }
+      })
+      .catch(() => {
+        this.logProviderLoading = false;
+      });
+
+    this.fileProviderLoading = true;
+    this.service
+      .getFileSystemNotificationProvider()
+      .then((fileNotificationProvider) => {
+        this.fileProviderLoading = false;
+        if (fileNotificationProvider.provider) {
+          this.fileNotificationProvider = fileNotificationProvider.provider;
+        }
+      })
+      .catch(() => {
+        this.fileProviderLoading = false;
+      });
```
remove as not needed

peintnermax commented on console/src/app/modules/policies/notification-smtp-provider/notification-smtp-provider.component.ts:
```diff
@@ -3,45 +3,38 @@ import { AbstractControl, UntypedFormBuilder, UntypedFormGroup } from '@angular/
+export class NotificationSMTPProviderComponent implements OnInit {
   @Input() public serviceType!: PolicyComponentServiceType;
-  public smsProviders: SMSProvider.AsObject[] = [];
   public logNotificationProvider!: DebugNotificationProvider.AsObject;
   public fileNotificationProvider!: DebugNotificationProvider.AsObject;
```
lets remove those providers are they are not displayed anyways..

peintnermax commented on console/src/app/modules/policies/notification-smtp-provider/notification-smtp-provider.component.ts:
```diff
@@ -3,45 +3,38 @@ import { AbstractControl, UntypedFormBuilder, UntypedFormGroup } from '@angular/
 
   public smtpLoading: boolean = false;
-  public smsProvidersLoading: boolean = false;
   public logProviderLoading: boolean = false;
   public fileProviderLoading: boolean = false;
```
rm 👍

doncicuto commented:
Thanks @peintnermax!, unused code has been removed

doncicuto commented:
Hi @peintnermax, I'll try to fix the conflicts and run e2e again in my local machine.

doncicuto commented on console/src/app/pages/org-settings/org-settings.component.ts:
```diff
@@ -68,7 +68,7 @@ export class OrgSettingsComponent implements OnInit {
   ngOnInit(): void {
     this.settingsList = this.authService
-      .isAllowedMapper(this.defaultSettingsList, (setting) => setting.requiredRoles.mgmt)
+      .isAllowedMapper(this.defaultSettingsList, (setting) => setting.requiredRoles.mgmt || [])
```
Hi @peintnermax, I've found that if we forget to add required roles to a setting that would cause an error, so if setting.requiredRoles.mgmt is undefined we'll pass an empty array to avoid that

peintnermax replied:
nice catch 👍

doncicuto commented on console/src/app/pages/instance-settings/instance-settings.component.ts:
```diff
@@ -80,7 +84,10 @@ export class InstanceSettingsComponent implements OnInit, OnDestroy {
-    this.settingsList = this.authService.isAllowedMapper(this.defaultSettingsList, (setting) => setting.requiredRoles.admin);
+    this.settingsList = this.authService.isAllowedMapper(
+      this.defaultSettingsList,
+      (setting) => setting.requiredRoles.admin || [],
```
Hi @peintnermax, I've found that if we forget to add required roles to a setting that would cause an error, so if setting.requiredRoles.mgmt is undefined we'll pass an empty array to avoid that

doncicuto commented:
Hi @peintnermax, I fixed the conflicts and the e2e tests. I've commented two lines that I've changed so you can review it. I've removed one of my new e2e tests because I was having an issue with cypress to check and uncheck a mat-checkbox.

Hope I didn't break things! If you need anything from me let me know.

peintnermax commented:
> Hi @peintnermax, I fixed the conflicts and the e2e tests. I've commented two lines that I've changed so you can review it. I've removed one of my new e2e tests because I was having an issue with cypress to check and uncheck a mat-checkbox.
> 
> Hope I didn't break things! If you need anything from me let me know.

amazing thx a lot 👍

peintnermax reviewed (approved):
amazing thx @doncicuto. Thx for updating the docs images too 🚀
